spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java

```diff
@@ -149,23 +153,53 @@ public final void onInit() {
 	protected void doInit() {
 	}
 
+
+	@Override
+	public void start() {
+		this.running.set(true);
+	}
+
+	@Override
+	public void stop() {
+		if (this.running.compareAndSet(true, false)) {
+			// remove unprocessed files from the queue (and filter)
+			AbstractFileInfo<F> file = this.toBeReceived.poll();
+			while (file != null) {
+				resetFilterIfnecessary(file);
+				file = this.toBeReceived.poll();
+			}
+		}
+	}
+
+	@Override
+	public boolean isRunning() {
+		return this.running.get();
+	}
+
 	@Override
 	protected Object doReceive() {
+		Assert.state(this.running.get(), getComponentName() + " is not running");
 		AbstractFileInfo<F> file = poll();
 		if (file != null) {
-			String remotePath = remotePath(file);
-			Session<?> session = this.remoteFileTemplate.getSession();
 			try {
-				return getMessageBuilderFactory()
-						.withPayload(session.readRaw(remotePath))
-						.setHeader(IntegrationMessageHeaderAccessor.CLOSEABLE_RESOURCE, session)
-						.setHeader(FileHeaders.REMOTE_DIRECTORY, file.getRemoteDirectory())
-						.setHeader(FileHeaders.REMOTE_FILE, file.getFilename())
-						.setHeader(FileHeaders.REMOTE_FILE_INFO,
-								this.fileInfoJson ? file.toJson() : file);
+				String remotePath = remotePath(file);
+				Session<?> session = this.remoteFileTemplate.getSession();
+				try {
+					return getMessageBuilderFactory()
+							.withPayload(session.readRaw(remotePath))
+							.setHeader(IntegrationMessageHeaderAccessor.CLOSEABLE_RESOURCE, session)
+							.setHeader(FileHeaders.REMOTE_DIRECTORY, file.getRemoteDirectory())
+							.setHeader(FileHeaders.REMOTE_FILE, file.getFilename())
+							.setHeader(FileHeaders.REMOTE_FILE_INFO,
+									this.fileInfoJson ? file.toJson() : file);
+				}
+				catch (IOException e) {
+					throw new MessagingException("IOException when retrieving " + remotePath, e);
```

Comment:
Can we change this to the `UncheckedIOException` instead? There is really nothing to do with the messaging...